metricbeat/helper/server/http/http_test.go

```diff
@@ -42,50 +111,115 @@ func GetHttpServer(host string, port int) (server.Server, error) {
 	}
 
 	httpServer := &http.Server{
-		Addr:    fmt.Sprintf("%s:%d", host, port),
+		Addr:    net.JoinHostPort(host, strconv.Itoa(int(port))),
 		Handler: http.HandlerFunc(h.handleFunc),
 	}
-	h.server = httpServer
+	if connectionType == "HTTPS" {
 
-	return h, nil
-}
+		certPem := []byte(`-----BEGIN CERTIFICATE-----
```

Comment:
what about addint the cert variables to the case table?

since we are using the same certificate pair for all tests, let's do something like
```
func TestHTTPServers(...) {
certPem := `yada yada yada` 
certKey := `more yada` 

var testCases = []struct {
		testName,
		certPem,
		certKey,
		"more stuff,
	}
```
What do you think?

Reply:
Just the HTTPS tests are using it, the HTTP tests does not use them. That's why i left them inside. I'm moving them to a separated function to clean the getHTTPServer